pkg/web/leaderboard.go

```diff
@@ -1,16 +1,79 @@
 package web
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
 	"strconv"
+	"strings"
+	"sync"
 
 	"github.com/nint8835/scribe/pkg/database"
 	"github.com/nint8835/scribe/pkg/web/ui/pages"
 )
 
 const QUOTES_PER_PAGE = 10
 
+// Cache to store input/output pairs
+var cache = make(map[string]string)
+var cacheMutex sync.Mutex // Mutex to handle concurrent access to the cache
+
+// GetGlobalName queries the DiscordLookup API and retrieves the global_name.
+func GetGlobalName(id string) (string, error) {
+	// Check if the result is already in the cache
+	cacheMutex.Lock()
+	if cachedName, found := cache[id]; found {
+		cacheMutex.Unlock()
+		return cachedName, nil
+	}
+	cacheMutex.Unlock()
+
+	// Construct the API URL
+	url := fmt.Sprintf("https://discordlookup.mesalytic.moe/v1/user/%s", id)
```

Comment:
scribe runs as a discord bot, you shouldn't need to use an external service to resolve usernames

Reply:
if you can get your hands on a discordgo.Session, you can do something similar to what riley has done here: https://github.com/nint8835/goldmark-discord-mentions/blob/main/resolver.go

Reply:
The bot's session is a globally-accessible object, you should just be able to use it here